provider/pkg/schema/gen.go

```diff
@@ -1210,37 +1212,70 @@ func (ctx *context) propertyTypeSpec(parentName string, propSchema *jsschema.Sch
 		}, nil
 	}
 
+	if propSchema.Items != nil && len(propSchema.Items.Schemas) > 0 {
+		elementType, err := ctx.propertyTypeSpec(parentName+"Item", propSchema.Items.Schemas[0])
+		if err != nil {
+			return nil, err
+		}
+		return &pschema.TypeSpec{
+			Type:  "array",
+			Items: elementType,
+		}, nil
+	}
+
+	if propSchema.Items != nil {
+		elementType, err := ctx.propertyTypeSpec(parentName+"Item", propSchema.Items.Schemas[0])
+		if err != nil {
+			return nil, err
+		}
+		return &pschema.TypeSpec{
+			Type:  "array",
+			Items: elementType,
+		}, nil
+	}
+
```

Comment:
I don't understand why you have both of these? it looks like the later is a strict superset of the former

Reply:
Yup, think this was a left-over from a refactoring that got reverted. Removed.